### PR TITLE
Support IF NOT EXISTS in create statement

### PR DESCRIPTION
**Pull request by dai-chen**
### Description

Add `IF NOT EXISTS` in create index statement. This is useful along with `checkpoint_location` option introduced in https://github.com/opensearch-project/opensearch-spark/pull/39. We just submit the exact same job (create with if not exists statement) to Spark to restart index refreshing, instead of submitting new job with `REFRESH` statement.

Doc: https://github.com/dai-chen/opensearch-spark/blob/support-if-not-exists/docs/index.md

### Issues Resolved

https://github.com/opensearch-project/opensearch-spark/issues/27

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSpark.scala [hunk 1]
```diff
@@ -74,14 +74,18 @@ class FlintSpark(val spark: SparkSession) {
    *
    * @param index
    *   Flint index to create
+   * @param ignoreIfExists
+   *   Ignore existing index
    */
-  def createIndex(index: FlintSparkIndex): Unit = {
+  def createIndex(index: FlintSparkIndex, ignoreIfExists: Boolean = false): Unit = {
     val indexName = index.name()
     if (flintClient.exists(indexName)) {
```
Comment (vamsi-amazon) [inline flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSpark.scala:82]: As per PR description, if we pass `IF NOT EXISTS` we perform refresh...where is that part? I could only see throw Exception.

**Comment by vamsi-amazon** [comment]
In general, what is preferred REFRESH INDEX statement or CREATE INDEX with IF NOT EXISTS.

File: flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSpark.scala (hunk 1 above)
Comment (dai-chen) [reply flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSpark.scala:82]: It's here in SQL command layer: https://github.com/opensearch-project/opensearch-spark/blob/main/flint-spark-integration/src/main/scala/org/opensearch/flint/spark/sql/skipping/FlintSparkSkippingIndexAstBuilder.scala#L53

By design, `FlintSpark` is on lower level which provides fine grained API (`createIndex` and `refreshIndex` separately) for higher level to combine them as wish.

**Comment by dai-chen** [comment]
> In general, what is preferred REFRESH INDEX statement or CREATE INDEX with IF NOT EXISTS.

I think run `REFRESH` statement directly is more clear. `CREATE INDEX with IF NOT EXISTS` is for use case like EMR job which we want to just restart with same job info without looking into it. Anyway, these options are only in SQL. In either case, only `FlintSpark.refreshIndex` API will be triggered underlying. Thanks!

**Review by vamsi-amazon — APPROVED** [review]

**Pull request merged**